batch-extension/BatchTensor.cpp

```diff
@@ -0,0 +1,97 @@
+#include "BatchTensor.h"
+
+
+static BatchTensorType& batch_tensor_singleton() {
+  static BatchTensorType value(&at::globalContext());
+  return value;
+}
+
+BatchTensor::BatchTensor(): TensorImpl(&batch_tensor_singleton()) {}
+
+BatchTensor::BatchTensor(at::Tensor data, at::Tensor mask, std::vector<bool> dims):
+TensorImpl(&batch_tensor_singleton()){
+  if(data.dim() != mask.dim() || mask.dim() != dims.size() + 1){
+    throw std::runtime_error("malformed MaskedBatch with " + std::string(data.toString()) + mask.toString());
+  }
+  data = data;
+  mask = mask;
+  dims = dims;
```

Comment:
Should probably do `dims = std::move(dims)` since you're already copying this vector by taking it by value